### PR TITLE
[SPARK-43621][PS][CONNECT] Enable `pyspark.pandas.spark.functions.repeat` in Spark Connect

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Enable `pyspark.pandas.spark.functions.repeat` in Spark Connect

### Why are the changes needed?
feature parity

### Does this PR introduce _any_ user-facing change?
yes, new method enabled

### How was this patch tested?
enabled UTs

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @itholic @HyukjinKwon

File: python/pyspark/pandas/spark/functions.py [hunk 1]
```diff
@@ -80,9 +80,22 @@ def repeat(col: Column, n: Union[int, Column]) -> Column:
...
+        return call_udf(  # type: ignore[return-value]
+            "repeat",
+            col,  # type: ignore[arg-type]
+            n,  # type: ignore[arg-type]
+        )
+
+    else:
+        sc = SparkContext._active_spark_context
+        n = _to_java_column(n) if isinstance(n, Column) else _create_column_from_literal(n)
+        return _call_udf(sc, "repeat", _to_java_column(col), n)
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/spark/functions.py:98]: actually I think we can just call `call_udf` instead without if-else.

Comment (zhengruifeng) [reply python/pyspark/pandas/spark/functions.py:98]: let me take another look

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**